web/src/components/dsprot/Transaction.vue

```diff
@@ -0,0 +1,42 @@
+<template>
+    <!-- CSS class depends on whether transaction is valid or not.
+    Property 'transaction' is used to determine this. -->
+    <div :class="[transaction.isValid ? 'valid' : 'invalid', 'transaction']">
+        <h3>ID: {{transaction.tid}}</h3>
+        <p>Points spent: {{transaction.userChoice}}</p>
+    </div>
+</template>
+
+<script>
+    export default {
+        name: 'TransactionUI',
+        props: {
+            transaction: Object
+        }
+    }
+</script>
+
+<style scoped>
+    .transaction h3 {
+        display: flex;
+        align-items: center;
+        justify-content: space-between;
+    }
+
+    .transaction p {
+        display: flex;
+        align-items: center;
+        justify-content: space-between;
+    }
+
+    .transaction {
+        background: #f4f4f4;
+        margin: 5px;
+        padding: 10px 20px;
+        border-left: 15px solid green;
+    }
+
+    .transaction.invalid {
+        border-left: 15px solid red;
+    }
```

Comment:
You could exchange this by tailwind stylings to have a more consistent look and feel

Reply:
have put that in issue #137 for now